### PR TITLE
WebPack and Browserify support for tiles, Icon.Canvas and Layer.Defferred.

### PR DESCRIPTION
**Pull request by FractalizeR**
WebPack and Browserify support for tiles, Icon.Canvas and Layer.Defferred.
Default leaflet constructor function provided where appropriate

### CODE REVIEW

**Comment by brunob** [comment]
I feel ok with default leaflet constructor function, but the other part is really needed ? I can't see things like that in leaflet fullscreen or other core plugins like markercluster.

**Comment by FractalizeR** [comment]
Well, if you are using Javascript modules, that would be helpful. I do not like globals much.

File: layer/Icon.Canvas.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-L.Icon.Canvas = L.Icon.extend({
+module.exports = L.Icon.Canvas = L.Icon.extend({
```
Comment (raphael-boucher) [inline layer/Icon.Canvas.js]: Won't will break in a browser that does not support es6 modules (currently most of them), since module is not defined ?

Comment (FractalizeR) [inline layer/Icon.Canvas.js]: Hmm... Probably. Should I fix it?

Comment (raphael-boucher) [inline layer/Icon.Canvas.js]: I offerred a solution in #178 :)

**Comment by FractalizeR** [comment]
Good ;)

**Comment by brunob** [comment]
@FractalizeR should we close this one and focus on #178 now ?

**Comment by FractalizeR** [comment]
@brunob  Yep, thanks!

**Pull request closed without merging**